src/plugins/intel_cpu/src/nodes/executors/kleidiai/kleidiai_mm.cpp

```diff
@@ -130,32 +171,74 @@ void MatMulKleidiAIExecutor::execute(const MemoryArgs& memory) {
     const size_t lhs_stride = K * sizeof(float);
     const size_t dst_stride_row = N * sizeof(float);
     const size_t dst_stride_col = sizeof(float);
-
     float* lhs = srcMem->getDataAs<float>();
-    float* rhs_packed = static_cast<float*>(rhsPackedMem->getData());
     float* dst = dstMem->getDataAs<float>();
-
+    
     size_t n_blocks = (N + BLOCK_SIZE - 1) / BLOCK_SIZE;
-    parallel_for(n_blocks, [&](size_t n_block) {
-        size_t n_start = (n_block * BLOCK_SIZE);
-        size_t n_end = std::min(n_start + BLOCK_SIZE, N);
-        size_t n_block_size = n_end - n_start;
-        const size_t rhs_packed_offset = ukernel.get_rhs_packed_offset(n_start, K);
-        const size_t dst_offset = ukernel.get_dst_offset(0, n_start, dst_stride_row);
-        const float* rhs_ptr = (rhs_packed + rhs_packed_offset / sizeof(float));
-        float* dst_ptr = (dst + dst_offset / (sizeof(float)));
-        ukernel.run_matmul(M,
-                           n_block_size,
-                           K,
-                           lhs,
-                           lhs_stride,
-                           rhs_ptr,
-                           dst_ptr,
-                           dst_stride_row,
-                           dst_stride_col,
-                           FLOAT_MIN,
-                           FLOAT_MAX);
-    });
+    
+    // Create packed LHS and RHS
+    if (inferPrecision == f32) {
+        float* rhs_packed = static_cast<float*>(rhsPackedMem->getData());
+
+        parallel_for(n_blocks, [&](size_t n_block) {
+            size_t n_start = (n_block * BLOCK_SIZE);
+            size_t n_end = std::min(n_start + BLOCK_SIZE, N);
+            size_t n_block_size = n_end - n_start;
+                const size_t rhs_packed_offset = ukernel_f32.get_rhs_packed_offset(n_start, K);
+                const size_t dst_offset = ukernel_f32.get_dst_offset(0, n_start, dst_stride_row);
+                const float* rhs_ptr = (rhs_packed + rhs_packed_offset / sizeof(float));
+                float* dst_ptr = (dst + dst_offset / (sizeof(float)));
+                ukernel_f32.run_matmul(
+                        M,
+                        n_block_size,
+                        K,
+                        lhs,
+                        lhs_stride,
+                        rhs_ptr,
+                        dst_ptr,
+                        dst_stride_row,
+                        dst_stride_col,
+                        FLOAT_MIN,
+                        FLOAT_MAX);
+        });
+    }
+    else if (inferPrecision == i8) {
+        const size_t lhsPackedSize = kai_get_lhs_packed_size_lhs_quant_pack_qai8dxp_f32(M, K, mr, kr, sr);
+        int8_t* lhs_packed_qa8dx = new int8_t[lhsPackedSize];
+        int8_t* rhs_packed_qs8cx = static_cast<int8_t*>(rhsPackedMem->getData());
```

Comment:
1. Same comment about the memory allocation. The only difference here is that memory size is dynamic (depends on dynamic shapes). In order to avoid reallocations on each inference I would recommend to use: createScratchPadMem utility to get memory of desired size (underligned logic will decide if reallocation is needed or already existing buffer can be reused): https://github.com/openvinotoolkit/openvino/blob/9968af6191e86d6b73ef9a7d988c43d01d942f3b/src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_fullyconnected.hpp#L78
2. Lets also move this allocation logic into MatMulKleidiAIExecutor::update function